From 03fffc35210207015f8353b17209ac086996496b Mon Sep 17 00:00:00 2001
From: Praveenkumar I <quic_ipkumar@quicinc.com>
Date: Fri, 15 Sep 2023 16:47:54 +0530
Subject: [PATCH 201/500] bus: mhi: host: Fix warning from dma_alloc_coherent()

mhi_cntrl->cntrl_dev should be used as device for dma_alloc_coherent().

Change-Id: I3cfd37381f7ca7f0f74e0b29880f7538bd33ad49
Signed-off-by: Praveenkumar I <quic_ipkumar@quicinc.com>
---
 drivers/bus/mhi/host/boot.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/bus/mhi/host/boot.c b/drivers/bus/mhi/host/boot.c
index e09b00216b0e..c23bbc327497 100644
--- a/drivers/bus/mhi/host/boot.c
+++ b/drivers/bus/mhi/host/boot.c
@@ -455,8 +455,8 @@ static int mhi_get_nonce(struct mhi_controller *mhi_cntrl)
 	mhi_read_reg(mhi_cntrl, mhi_cntrl->regs, PCIE_PCIE_LOCAL_REG_PCIE_LOCAL_RSV1,
 			&sram_addr);
 	if (sram_addr != 0) {
-		mhi_cntrl->nonce_buf = dma_alloc_coherent(dev, NONCE_SIZE,
-					&mhi_cntrl->nonce_dma_addr, GFP_KERNEL);
+		mhi_cntrl->nonce_buf = dma_alloc_coherent(mhi_cntrl->cntrl_dev, NONCE_SIZE,
+							  &mhi_cntrl->nonce_dma_addr, GFP_KERNEL);
 		if (!mhi_cntrl->nonce_buf) {
 			dev_err(dev, "Error Allocating memory buffer for NONCE\n");
 			return -ENOMEM;
-- 
2.34.1

